src/prefect/engine.py

```diff
@@ -1387,6 +1402,8 @@ async def submit_task_run(
             ),
             log_prints=should_log_prints(task),
             settings=prefect.context.SettingsContext.get().copy(),
+            user_thread=user_thread,
+            concurrency_type=task_runner.concurrency_type,
```

Comment:
We need the concurrency type so we can check if the task run is sequential for scheduling on the main thread. We can't use `flow_run_context.task_runner.concurrency_type` because we use a different task runner for sequential task calls.